runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/model/TestConverterHandler.java

```diff
@@ -46,6 +56,19 @@ public TestConverterHandler(
         schema = cataloged != null ? cataloged.schemas.get(0) : null;
         schemaId = schema != null ? schema.id : 0;
         this.handler = cataloged != null ? supplyCatalog.apply(cataloged.id) : null;
+        this.pattern = Pattern.compile(PATH);
+        this.extracted = new HashMap<>();
+    }
+
+    @Override
+    public void extract(
+        String path)
+    {
+        Matcher matcher = pattern.matcher(path);
```

Comment:
Let's reuse the matcher, via precreating with `pattern.matcher("")` and then `matcher.reset(path)` here.

Same matcher reuse for extractedLength below.